docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,567 @@
+   [how to setup your fork](#setting-up-your-fork) in a subsection below.
+3. Double-check that `swift`'s sibling directories are present.
+   ```sh
+   ls ..
+   ```
+   This should list directories like `llvm-project`, `swiftpm` and so on.
+4. Checkout the right branch/tag:
+   If you are building the toolchain for local development, you can skip this
+   step, as Step 2 will checkout `swift`'s `master` branch and matching
+   branches for other projects. <!-- [TODO: master-to-main-rename] -->
+   If you are building the toolchain as a one-off, it is more likely that you
+   want a specific branch or a tag, often corresponding to a specific release
+   or a specific snapshot. You can update the branch/tag for all repositories
+   as follows:
+   ```sh
+   utils/update-checkout --branch mybranchname
+   # OR
+   utils/update-checkout --tag mytagname
+   ```
+   Detailed branching information, including names for release branches, can
+   be found in [Branches.md](/docs/Branches.md).
+
+**Note:**
+The commands used in the rest of this guide assumes that the absolute path
+to your working directory is something like `/path/to/swift-project/swift`.
+Double-check that running `pwd` prints a path ending with `swift`.
+
+[uploaded your SSH keys to GitHub]: https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/
+
+### Setting up your fork
+
+If you are building the toolchain for development and submitting patches,
+you will need to setup a GitHub fork.
+
+First fork the `apple/swift` [repository](https://github.com/apple/swift.git),
+using the "Fork" button in the web UI, near the top-right. This will create a
+repository `username/swift` for your GitHub username. Next, add it as a remote:
+```sh
+# Using 'my-remote' as a placeholder name.
+
+# If you set up SSH in step 2
+git remote add my-remote git@github.com:username/swift.git
+
+# If you used HTTPS in step 2
+git remote add my-remote https://github.com/username/swift.git
+```
+Finally, create a new branch.
+```sh
+# Using 'my-branch' as a placeholder name
+git checkout my-branch
+git push --set-upstream my-remote my-branch
+```
+
+### Troubleshooting cloning issues
+
+- If `update-checkout` failed, double-check that the absolute path to your
+  working directory does not have non-ASCII characters.
+- If `update-checkout` failed and the absolute path to your working directory
+  had spaces in it, please [file a bug report][Swift JIRA] and change the path
+  to work around it.
+- Before running `update-checkout`, double-check that `swift` is the only
+  repository inside the `swift-project` directory. Otherwise,
+  `update-checkout` may not clone the necessary dependencies.
+
+## Installing dependencies
+
+### macOS
+
+1. Install [Xcode 12 beta 3][Xcode] or newer:
+   The required version of Xcode changes frequently and is often a beta release.
+   Check this document or the host information on <https://ci.swift.org> for the
+   current required version.
```

Comment:
It would be ideal if an Xcode version check happened within scope of `build-script`, to fail early and report the issue in a clear way.

I've been bit before: using the wrong Xcode still kicks off a build, and then some point in the build, maybe a while later, a compilation error happens. The error is not likely to point to Xcode version as the root cause, which leads to searches for keywords found in the error message. If the person is lucky, someone else has already hit the error, posted on the forums, and received an answer saying they need to use a newer version of Xcode.

Reply:
I think there are two distinct potential failure modes here:

1. The documentation is up-to-date but you're using an older Xcode (you either didn't see that the documentation was updated or something else).
2. The documentation is out-of-date.

Having `build-script` check this would address failure mode 1 but not 2 -- someone might just forget to update `build-script` instead of forgetting to update the documentation.

That said, I do think it would be a valuable addition to have as a follow-up (not as part of this PR) because it at least addresses one failure mode.

Reply:
Filed https://bugs.swift.org/browse/SR-13497.